efi/secureboot_namespace_rules.go

```diff
@@ -90,7 +126,10 @@ func (r *secureBootNamespaceRules) AddAuthorities(certs ...*x509.Certificate) {
 		for _, authority := range r.authorities {
 			if bytes.Equal(authority.subject, cert.RawSubject) &&
 				bytes.Equal(authority.subjectKeyId, cert.SubjectKeyId) &&
-				authority.publicKeyAlgorithm == cert.PublicKeyAlgorithm {
+				authority.publicKeyAlgorithm == cert.PublicKeyAlgorithm &&
+				bytes.Equal(authority.issuer, cert.RawIssuer) &&
+				bytes.Equal(authority.authorityKeyId, cert.AuthorityKeyId) &&
+				authority.signatureAlgorithm == cert.SignatureAlgorithm {
```

Comment:
One side will always have all fields. I did consider trying to accommodate this but it looked over complicated and it would only make a difference in an edge case - ie, a shim that embeds the Microsoft CA as a vendor certificate. In this case, it would be added twice, which isn't really a problem anyway.